src/Compilers/Core/CommandLine/ConsoleUtil.cs

```diff
@@ -41,7 +38,7 @@ internal static T RunWithUtf8Output<T>(Func<TextWriter, T> func)
 
         internal static T RunWithUtf8Output<T>(bool utf8Output, TextWriter textWriter, Func<TextWriter, T> func)
         {
-            if (utf8Output)
+            if (utf8Output && textWriter.Encoding.CodePage != s_utf8Encoding.CodePage)
             {
                 if (textWriter != Console.Out)
```

Comment:
It seems like we now believe this case to be invalid -- namely that it's OK to pass `utf8output` even if not writing to `Console.Out`. I'm not sure that's wrong, but I'd like to understand if we took any dependencies on that fact.

Reply:
(Spoke in-person on Friday, this change is okay)